controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -93,46 +89,56 @@ public void refresh() {
     }
 
     @Override
-    CompletableFuture<Void> checkStreamExists(Create create) {
-        CompletableFuture<Void> result = new CompletableFuture<>();
+    CompletableFuture<CreateStreamResponse> checkStreamExists(StreamConfiguration configuration, long timestamp) {
+        CompletableFuture<CreateStreamResponse> result = new CompletableFuture<>();
 
         long time = creationTime.get();
-        if (time != Long.MIN_VALUE && create.getCreationTime() != time) {
-            result.completeExceptionally(StoreException.create(StoreException.Type.DATA_EXISTS, getName()));
-        } else {
-            result.complete(null);
+        synchronized (lock) {
```

Comment:
> hence we acquire lock in this method too for reading configuration and state

could we just read configuration and state under lock and do the rest of the processing outside the synchronized block or does the processing also need to be synchronized?